lib/scanner/walk.go

```diff
@@ -124,6 +127,16 @@ func (w *walker) walk(ctx context.Context) chan ScanResult {
 		close(toHashChan)
 	}()
 
+	if w.DetectChangesOnly {
+		go func() {
+			for file := range toHashChan {
+				finishedChan <- ScanResult{File: file}
+			}
+			close(finishedChan)
+		}()
+		return finishedChan
+	}
+
```

Comment:
This feels like a hack: a config option which essentially changes what the function is doing (much less than without the config option). Perhaps `walk` could be refactored to a few functions, where the function name, not the config option, is setting what the caller is expecting.

Reply:
I introduced `WalkWithoutHashing` in https://github.com/syncthing/syncthing/pull/6727/commits/faa39d179385df44f03e193cb6b90e764ae6d414  
There wasn't much refactoring necessary, as code for walking the filesystem and detecting changes was already well separated from hashing files. The previous solution with the config option needed a single condition with 7 lines of code in it to trigger the different behaviour (those 7 lines now live in `walkWithoutHashing`).